sdk/helper/keysutil/policy.go

```diff
@@ -1163,6 +1170,37 @@ func (p *Policy) Sign(ver int, context, input []byte, hashAlgorithm HashType, si
 			return nil, errutil.InternalError{Err: fmt.Sprintf("unsupported rsa signature algorithm %s", sigAlgorithm)}
 		}
 
+	case KeyType_ECDSA_secp256k1:
+		switch hashAlgorithm {
+		case HashTypeKeccak256:
+		default:
+			return nil, errutil.InternalError{Err: "unsupported hash algorithm"}
+		}
+
+		key := keyParams.Key
+		hexKey := hexutil.Encode(key)[2:]
+
+		privateKey, err := ethcrypto.HexToECDSA(hexKey)
+		if err != nil {
+			return nil, err
+		}
+
+		hash := ethcrypto.Keccak256Hash(input)
```

Comment:
This not working for bitcoin-based, keccak hash work only for ethereum network. For bitcoin-base double sha256.

Reply:
Correct, I’m working on Bitcoin now, this implementation for Ethereum only

Reply:
It just so happened that a week ago we did this task for ether and bitcoin as a separate plugin, let's help :)

Reply:
`entry.FormattedPublicKey` this field is not universal, in the same bitcoin networks there are different formats, it is better not to make a rigid connection to a specific network.

If you remove hash at the initial stage, and receive already hashed data in transit, then there will be a universal solution for any cryptocurrency.

Reply:
and the `hexutil` from `go-ethereum` use the `0x` prefix, which is not required in bitcoin-based.

Reply:
and the `hexutil` from `go-ethereum` use the `0x` prefix, which is not required in bitcoin-based.

Reply:
> It just so happened that a week ago we did this task for eth and bitcoin as a separate plugin, let's help :)

@intech is the code to the separate plugin somewhere on github? seem's like an interesting one to use if available already

Reply:
Got your point about "prehashed", that makes sense, thanks, doing changes now